pyhelios/util/read_obj.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+import open3d as o3d
+import os
+
+def read_obj(path, logging=False):
+    import open3d as o3d
+    import os
+
+    # Read file into memory
+    with open(path, 'r') as f:
+            lines = f.readlines()
+
+    # Count number of faces with more than 3 vertices and overall maximum number of vertices
+    count=0
+    max_vert_num = 0
+    four_verts = False
+    for line in lines:
+        line = line.split(' ')
+    if line[0] == 'f' and len(line) > 4:
+        count += 1
+        max_vert_num = len(line)-1 if len(line)-1 > max_vert_num else max_vert_num
+    if line[0] == 'f' and len(line) == 5:
+        four_verts = True
+    
+    if logging:
+        print('Total number of faces in OBJ with more than 3 vertices: {}'.format(count))
+        print('Total number of faces in OBJ with 4 vertices: {}'.format(count))
+        print('Maximum number of vertices per face: {}'.format(max_vert_num))
+    
+    if max_vert_num > 4:
+        print('[HELIOS.PY WARNING] {} contains at least one face with more than 4 vertices. These faces will not be visualised and are not considered by HELIOS++'.format(path))
+        
+    # If a face with four vertices exists, create copy of obj and write modified faces to new obj
+    if four_verts:
+        copy = path.replace('.obj', '_copy.obj')
```

Comment:
the input `.obj` could be in a location that is non-writable. If you use the `tempfile` package, you can ensure that you get a path location that is writable by the user:
https://docs.python.org/3/library/tempfile.html